osu.Game.Tests/Editing/EditorChangeHandlerTest.cs

```diff
@@ -44,6 +43,36 @@ public void TestSaveRestoreState()
             Assert.That(stateChangedFired, Is.EqualTo(2));
         }
 
+        [Test]
+        public void TestApplyThenUndoThenApplySameChange()
+        {
+            var (handler, beatmap) = createChangeHandler();
+
+            Assert.That(handler.CanUndo.Value, Is.False);
+            Assert.That(handler.CanRedo.Value, Is.False);
+
+            string originalHash = handler.CurrentStateHash;
+
+            addArbitraryChange(beatmap);
+            handler.SaveState();
+
+            Assert.That(handler.CanUndo.Value, Is.True);
+            Assert.That(handler.CanRedo.Value, Is.False);
+            Assert.That(stateChangedFired, Is.EqualTo(1));
+
+            string hash = handler.CurrentStateHash;
+
+            // save a save without making any changes
```

Comment:
can't seem to decipher this statement - did you mean something like:

```suggestion
            // undo a change without saving
```